### PR TITLE
docs: insert remove data path function

### PR DESCRIPTION
**Pull request by rokamu623**
## Description
The docstring of the function to insert and remove the data path is described.

The functions covered are as follows:
`update_message_context`
`insert_publisher_callback`
`insert_variable_passing`
`remove_publisher_callback`
`remove_variable_passing`

## Related links
Jira: https://tier4.atlassian.net/browse/RT2-400
github: https://github.com/tier4/CARET_analyze/pull/297
github: https://github.com/tier4/CARET_analyze/pull/196

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [ ] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/architecture/architecture.py [hunk 1]
```diff
@@ -356,6 +356,21 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
...
+        Update message_context of node_path in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node to update message_context
+        context_type : str
+            type name of updating message_context
+        subscribe_topic_name : str
+            name of subscribe topic of node_path updating message_context
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:369]:
```suggestion
            name of subscribe topic of node_path for updating message_context
```

File: src/caret_analyze/architecture/architecture.py [hunk 2]
```diff
@@ -356,6 +356,21 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
...
+        publish_topic_name : str
+            name of publish topic of node_path updating message_context
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:371]:
```suggestion
            name of publish topic of node_path updating for message_context
```

File: src/caret_analyze/architecture/architecture.py [hunk 3]
```diff
@@ -381,6 +409,19 @@ def insert_publisher_callback(self, node_name: str,
...
+        Insert variable passing in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node to insert variable passing
+        callback_name_write : str
+            name of write callback in variable passing to insert
+        callback_name_read : str
+            name of read callback in variable passing to insert
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:422]:
```suggestion
        Insert variable_passing in "node_name" node.

        Parameters
        ----------
        node_name : str
            name of target node to insert variable_passing
        callback_name_write : str
            name of write callback in variable_passing to insert
        callback_name_read : str
            name of read callback in variable_passing to insert
```

File: src/caret_analyze/architecture/architecture.py [hunk 4]
```diff
@@ -399,6 +453,19 @@ def remove_publisher_callback(self, node_name: str,
...
+        Remove variable passing in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node to remove variable passing
+        callback_name_write : str
+            name of write callback in variable passing to remove
+        callback_name_read : str
+            name of read callback in variable passing to remove
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:466]:
```suggestion
        Remove variable_passing in "node_name" node.

        Parameters
        ----------
        node_name : str
            name of target node to remove variable_passing
        callback_name_write : str
            name of write callback in variable_passing to remove
        callback_name_read : str
            name of read callback in variable_passing to remove
```

File: src/caret_analyze/architecture/architecture.py [hunk 5]
```diff
@@ -390,6 +431,19 @@ def insert_variable_passing(self, node_name: str,
 
     def remove_publisher_callback(self, node_name: str,
                                   publish_topic_name: str, callback_name: str) -> None:
+        """
+        Remove callback to publisher in "node_name" node.
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:435]: Please clarify whether this function to remove the publisher's callback or both the publisher and its callback.

File: src/caret_analyze/architecture/architecture.py [hunk 6]
```diff
@@ -372,6 +387,19 @@ def update_message_context(self, node_name: str, context_type: str,
...
+        Insert callback to publisher in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node to insert publisher and callback
+        publish_topic_name : str
+            name of publish topic to insert publisher callback
+        callback_name : str
+            name of callback to insert publisher
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:400]: name of callback to be inserted for publisher?

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: src/caret_analyze/architecture/architecture.py (hunk 1 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [c5a7bd9](https://github.com/tier4/CARET_analyze/pull/300/commits/c5a7bd978be6ff8d12c907120a79795467a95842).

File: src/caret_analyze/architecture/architecture.py (hunk 2 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [c5a7bd9](https://github.com/tier4/CARET_analyze/pull/300/commits/c5a7bd978be6ff8d12c907120a79795467a95842).

File: src/caret_analyze/architecture/architecture.py (hunk 3 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [c5a7bd9](https://github.com/tier4/CARET_analyze/pull/300/commits/c5a7bd978be6ff8d12c907120a79795467a95842).

File: src/caret_analyze/architecture/architecture.py (hunk 4 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [c5a7bd9](https://github.com/tier4/CARET_analyze/pull/300/commits/c5a7bd978be6ff8d12c907120a79795467a95842).

File: src/caret_analyze/architecture/architecture.py (hunk 5 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: This function deletes "which CB is registered with the publisher".

File: src/caret_analyze/architecture/architecture.py (hunk 6 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [474c2bc](https://github.com/tier4/CARET_analyze/pull/300/commits/474c2bcf9fff70e85d47dc912f091e3c53122542).

File: src/caret_analyze/architecture/architecture.py [hunk 7]
```diff
@@ -381,6 +409,19 @@ def insert_publisher_callback(self, node_name: str,
...
+        Insert variable_passing in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node
+        callback_name_write : str
+            name of write callback in variable_passing to insert
+        callback_name_read : str
+            name of read callback in variable_passing to insert
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:422]:
```suggestion
            name of write callback to be inserted in variable_passing
        callback_name_read : str
            name of read callback to be inserted in variable_passing
```

File: src/caret_analyze/architecture/architecture.py [hunk 8]
```diff
@@ -372,6 +387,19 @@ def update_message_context(self, node_name: str, context_type: str,
 
     def insert_publisher_callback(self, node_name: str,
                                   publish_topic_name: str, callback_name: str) -> None:
+        """
+        Insert the association of callback with publisher in "node_name" node.
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:391]: There is no rule as to whether an article should be added or not, but it should be consistent.

```suggestion
        Insert association of callback with publisher in "node_name" node.
```

File: src/caret_analyze/architecture/architecture.py [hunk 9]
```diff
@@ -356,6 +356,21 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
...
                                subscribe_topic_name: str, publish_topic_name: str) -> None:
+        """
+        Update message_context of node_path in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node
+        context_type : str
+            type name of updating message_context
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:367]:
```suggestion
            type name of message_context to be added
```

File: src/caret_analyze/architecture/architecture.py [hunk 10]
```diff
@@ -372,6 +387,19 @@ def update_message_context(self, node_name: str, context_type: str,
...
                                   publish_topic_name: str, callback_name: str) -> None:
+        """
+        Insert the association of callback with publisher in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node
+        publish_topic_name : str
+            name of publish topic to insert publisher callback
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:398]:
```suggestion
            topic name of target publisher into which callback is inserted
```

File: src/caret_analyze/architecture/architecture.py [hunk 11]
```diff
@@ -390,6 +431,19 @@ def insert_variable_passing(self, node_name: str,
 
     def remove_publisher_callback(self, node_name: str,
                                   publish_topic_name: str, callback_name: str) -> None:
+        """
+        Remove the association of callback with publisher in "node_name" node.
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:435]:
```suggestion
        Remove association of callback with publisher in "node_name" node.
```

File: src/caret_analyze/architecture/architecture.py [hunk 12]
```diff
@@ -390,6 +431,19 @@ def insert_variable_passing(self, node_name: str,
...
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node
+        publish_topic_name : str
+            name of publish topic to remove publisher callback
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:442]:
```suggestion
            topic name of target publisher from which callback is deleted
```

File: src/caret_analyze/architecture/architecture.py [hunk 13]
```diff
@@ -399,6 +453,19 @@ def remove_publisher_callback(self, node_name: str,
...
+        Remove variable_passing in "node_name" node.
+
+        Parameters
+        ----------
+        node_name : str
+            name of target node
+        callback_name_write : str
+            name of write callback in variable_passing to remove
+        callback_name_read : str
+            name of read callback in variable_passing to remove
```
Comment (atsushi421) [inline src/caret_analyze/architecture/architecture.py:466]:
```suggestion
            name of write callback to be removed from variable_passing
        callback_name_read : str
            name of read callback to be removed from variable_passing
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: src/caret_analyze/architecture/architecture.py (hunk 13 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 12 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 11 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 10 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 9 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 8 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

File: src/caret_analyze/architecture/architecture.py (hunk 7 above)
Comment (rokamu623) [reply src/caret_analyze/architecture/architecture.py]: Fix it in [6cf0285](https://github.com/tier4/CARET_analyze/pull/300/commits/6cf0285fa738613d5ed20fe836eed60ce682205c).

**Comment by rokamu623** [comment]
@atsushi421
Review reflected. Please reconfirm.

**Review by atsushi421 — APPROVED** [review]
LGTM

**Comment by rokamu623** [comment]
@nabetetsu 
Please review.

**Review by nabetetsu — APPROVED** [review]
LGTM :+1: 

No action is required here, but we would like to make the argument names and docstring contents more user-friendly.

**Pull request merged**